src/main/java/com/takipi/oss/storage/resources/fs/multifetcher/S3CacheImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.takipi.oss.storage.resources.fs.multifetcher;
+
+import com.takipi.oss.storage.caching.Cache;
+import com.takipi.oss.storage.caching.InMemoryCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3CacheImpl implements S3Cache {
+
+    private static final Logger logger = LoggerFactory.getLogger(S3CacheImpl.class);
+
+    private final InMemoryCache memoryCache;
+    private final Cache cache;
+
+    public S3CacheImpl(long maxSize) {
+        memoryCache = new InMemoryCache.Builder().setMaxSize(maxSize).build();
+        cache = new Cache(memoryCache);
```

Comment:
Allow to add CacheLogger depending on the configuration

Reply:
Currently we just use an in memory cache. We can always change it in future.